### PR TITLE
Addresses style issues

### PR DESCRIPTION
**Pull request by trinchan**
Addresses warnings for
golint
go vet
goimports
gofmt
ineffassign

aggregated through https://goreportcard.com/report/github.com/compose/transporter

also adds goreportcard badge to the readme

### CODE REVIEW

File: pkg/adaptor/elasticsearch/elasticsearch.go [hunk 1]
```diff
@@ -12,9 +12,9 @@ import (
 	elastigo "github.com/mattbaird/elastigo/lib"
 )
 
-// Elasticsearch is an adaptor to connect a pipeline to
+// ElasticSearch is an adaptor to connect a pipeline to
 // an elasticsearch cluster.
-type Elasticsearch struct {
+type ElasticSearch struct {
```
Comment (nstott) [inline pkg/adaptor/elasticsearch/elasticsearch.go]: this one might get people in a tizzy
the name is actually Elasticsearch, not ElasticSearch

**Comment by nstott** [comment]
nice!

File: pkg/adaptor/elasticsearch/elasticsearch.go (hunk 1 above)
Comment (jipperinbham) [inline pkg/adaptor/elasticsearch/elasticsearch.go]: ha, yea, probably best to keep it as not CamelCase

Comment (trinchan) [inline pkg/adaptor/elasticsearch/elasticsearch.go]: ah, my bad. i thought it was Elastic Search by the Elastic company. fixed!

File: pkg/adaptor/elasticsearch/elasticsearch.go [hunk 2]
```diff
@@ -12,9 +12,9 @@ import (
 	elastigo "github.com/mattbaird/elastigo/lib"
 )
 
-// Elasticsearch is an adaptor to connect a pipeline to
+// Elasticsearcb is an adaptor to connect a pipeline to
 // an elasticsearch cluster.
-type Elasticsearch struct {
+type Elasticsearcb struct {
```
Comment (jipperinbham) [inline pkg/adaptor/elasticsearch/elasticsearch.go]: typo :fearful:

**Comment by jipperinbham** [comment]
nice cleanup! :+1:

File: pkg/adaptor/elasticsearch/elasticsearch.go (hunk 2 above)
Comment (trinchan) [inline pkg/adaptor/elasticsearch/elasticsearch.go]: you saw nothing :jedi:

**Comment by trinchan** [comment]
this is just a style cleanup. i'm working on a more fundamental refactor that'll be in a separate PR

**Pull request merged**